### DISCUSSION TITLE
No delay in the result predictions despite putting a 366 day sequence length

### DISCUSSION THREAD

**Discussion by justagoodguy-44** [neuralhydrology/neuralhydrology#173] [Q&A] [answered]
Hi, thanks a lot for making this repo available!
There is just something I think I'm not understanding. In the config I define the seq_length = 366 days and predict_last_n = 1, so I'd expect there to be 365 days with no prediction at the start of the result on the test set. Yet for both the cudaLSTM and the forecast-handoff models it would seem that I obtain predictions starting from the first day of the test set. Any idea on why this is the case?

Thanks a lot,
Paul

**Comment by gauchm** [comment, marked as answer]
The reason is [here](https://github.com/neuralhydrology/neuralhydrology/blob/master/neuralhydrology/datasetzoo/basedataset.py#L409). The dataset adds an offset to start the input sequence 365 days earlier, so that the first valid prediction will happen on the first day of your test set.

**Reply by justagoodguy-44** [reply]
Oh I see, thanks a lot! Does this mean I should make sure that the 365 days before the start of the test set are not used either in the train or validation set?

**Reply by gauchm** [reply]
I guess that's debatable and depends on how strict you want to be. The model will never see _discharge_ (i.e., labels) from the wrong period, but it will see _inputs_ from other periods.
Generally speaking, the results for val/test date ranges close to the training period will definitely be positively biased. So if your goal is to use the val/test results to gauge how good your model will behave in the future, some padding may be a good idea. But it's also a tradeoff since you don't want to lose too much val/test data (you could avoid losing val/test time with a k-fold cross-validation scheme, though).

**Reply by justagoodguy-44** [reply]
I understand now, thank you very much!
